lib/std/heap/binned_allocator.zig

```diff
@@ -0,0 +1,818 @@
+//! # Binned Allocator
+//!
+//! This is designed to be a fast, general-purpose allocator, for use in place of GeneralPurposeAllocator when speed is the priority.
+//! It uses a collection of "bins" for various sizes of allocation, combined with a binary tree for allocations larger than 4kiB.
+//!
+//! Features:
+//!  - zero per-allocation metadata
+//!  - optional thread-safety (on by default; disable for improved single-threaded performance)
+//!  - primitive leak checking (for full checking, use GeneralPurposeAllocator)
+//!  - does not free to backing allocator, allowing use with FixedBufferAllocator
+//!
+//! Limitations:
+//!  - allocations will be at least as big as their alignment
+//!  - allocation sizes will be rounded up to the nearest bin size
+//!  - alignment cannot be greater than std.mem.page_size (see also #7952)
+//!  - does not free to backing allocator, leading to memory leaks for some niche allocation patterns
+
+const std = @import("std");
+const builtin = @import("builtin");
+
+pub const Config = struct {
+    /// Whether to synchronize usage of this allocator.
+    /// For actual thread safety, the backing allocator must also be thread safe.
+    thread_safe: bool = !builtin.single_threaded,
+
+    /// Whether to warn about leaked memory on deinit.
```

Comment:
This sounds like it does not set return status, but prints to stderr.

Reply:
That is correct